core/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -648,6 +657,144 @@ public boolean equals(Object other) {
+      this.attrs = checkNotNull(attrs, "attrs");
+      this.stateListener = checkNotNull(stateListener, "SubchannelStateListener is not set");
+    }
+
+    /**
+     * Returns the addresses, which is an unmodifiable list.
+     */
+    public List<EquivalentAddressGroup> getAddresses() {
+      return addrs;
+    }
+
+    /**
+     * Returns the attributes.
+     */
+    public Attributes getAttributes() {
+      return attrs;
+    }
+
+    /**
+     * Returns the state listener.
+     */
+    public SubchannelStateListener getStateListener() {
+      return stateListener;
+    }
+
+    /**
+     * Returns a builder with the same initial values as this object.
+     */
+    public Builder toBuilder() {
+      return newBuilder().setAddresses(addrs).setAttributes(attrs).setStateListener(stateListener);
+    }
+
+    /**
+     * Creates a new builder.
+     */
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public static final class Builder {
+      private List<EquivalentAddressGroup> addrs;
+      private Attributes attrs = Attributes.EMPTY;
+      private SubchannelStateListener stateListener;
+
+      Builder() {
+      }
+
+      /**
+       * The addresses to connect to.  All addresses are considered equivalent and will be tried
+       * in the order they are provided.
+       */
+      public Builder setAddresses(EquivalentAddressGroup addrs) {
+        this.addrs = Collections.singletonList(addrs);
+        return this;
+      }
+
+      /**
+       * Required.  The addresses to connect to.  All addresses are considered equivalent and will
+       * be tried in the order they are provided.
+       *
+       * @throws IllegalArgumentException if {@code addrs} is empty
+       */
+      public Builder setAddresses(List<EquivalentAddressGroup> addrs) {
+        checkArgument(!addrs.isEmpty(), "addrs is empty");
+        this.addrs = Collections.unmodifiableList(new ArrayList<>(addrs));
+        return this;
+      }
+      
+      /**
+       * Optional. Attributes provided here will be included in {@link Subchannel#getAttributes}.
+       * Default is empty.
+       */
+      public Builder setAttributes(Attributes attrs) {
+        this.attrs = checkNotNull(attrs, "attrs");
+        return this;
+      }
+
+      /**
+       * Required.  Receives state changes of the created Subchannel.  The listener is called from
+       * the {@link #getSynchronizationContext Synchronization Context}.  It's safe to share the
+       * listener among multiple Subchannels.
+       */
+      public Builder setStateListener(SubchannelStateListener listener) {
+        this.stateListener = checkNotNull(listener, "listener");
+        return this;
+      }
+
+      /**
+       * Creates a new args object.
+       */
+      public CreateSubchannelArgs build() {
+        return new CreateSubchannelArgs(addrs, attrs, stateListener);
+      }
+    }
+
+    @Override
+    public String toString() {
```

Comment:
It'd be easier to read this class if these methods weren't split by the Builder class in-between.